common/src/chain/transaction/output.rs

```diff
@@ -41,26 +41,33 @@ pub enum OutputPurpose {
     #[codec(index = 1)]
     LockThenTransfer(Destination, OutputTimeLock),
     #[codec(index = 2)]
-    StakePool(Box<StakePoolData>),
-    #[codec(index = 3)]
     Burn,
+    /// Output type that is used to create a stake pool
+    #[codec(index = 3)]
+    StakePool(Box<StakePoolData>),
+    /// Output type that represents spending of a stake pool output in a block reward
+    /// in order to produce a block
+    #[codec(index = 4)]
+    SpendStakePool(Box<StakePoolData>, OutputTimeLock),
```

Comment:
I wouldn't call this "Spend"... Maybe something like `StakeToProduceBlock`? I'm open to suggestions here.

Reply:
Just to be clear, "spending" is the opposite of having an output available.

Reply:
> Just to be clear, "spending" is the opposite of having an output available.

I'd say that burning is the opposite but spending is always using an output and producing another. So here I spend StakePool and produce SpendStakePool.

But in general I agree that the name can be improved. I searched for the verb that better describes the intention. Maybe the purpose of the output is something like `UtiliseStakePool`/`ApplyStakePool`/`RecycleStakePool`?

Reply:
In `StakeToProduceBlock` haven't we already staked when a pool was created? The meaning of stake verb is a bit vague to me